data_hub_api/kotahi_docmaps/v1/codecs/docmaps.py

```diff
@@ -0,0 +1,210 @@
+import logging
+import ast
+from typing import Dict, Iterable, Sequence, cast
+import urllib
+
+from data_hub_api.kotahi_docmaps.v1.codecs.elife_manuscript import (
+    get_docmap_elife_manuscript_doi_assertion_item,
+    get_docmap_elife_manuscript_doi_assertion_item_for_vor,
+    get_docmap_elife_manuscript_output,
+)
+from data_hub_api.kotahi_docmaps.v1.codecs.evaluation import (
+    iter_docmap_actions_for_evaluations
+)
+from data_hub_api.kotahi_docmaps.v1.codecs.preprint import (
+    get_docmap_preprint_assertion_item,
+    get_docmap_preprint_input,
+    get_docmap_preprint_input_with_published
+)
+from data_hub_api.kotahi_docmaps.v1.api_input_typing import ApiInput, ApiManuscriptVersionInput
+
+from data_hub_api.kotahi_docmaps.v1.docmap_typing import (
+    DocmapAction,
+    DocmapAssertion,
+    DocmapStep,
+    DocmapSteps,
+    Docmap
+)
+
+from data_hub_api.utils.json import remove_key_with_none_value_only
+
+
+LOGGER = logging.getLogger(__name__)
+
+DOCMAPS_JSONLD_SCHEMA_URL = 'https://w3id.org/docmaps/context.jsonld'
+
+DOCMAP_ID_PREFIX = (
+    'https://data-hub-api.elifesciences.org/kotahi/docmaps/v1/'
+    +
+    'by-publisher/elife/get-by-manuscript-id?'
+)
+
+
+def get_docmap_assertions_for_under_review_step(
+    query_result_item: ApiInput,
+    manuscript_version: ApiManuscriptVersionInput
+) -> Sequence[DocmapAssertion]:
+    return [{
+        'item': get_docmap_preprint_assertion_item(manuscript_version=manuscript_version),
+        'status': 'under-review',
+        'happened': (
```

Comment:
This looks like a function worth extracting.

e.g. `get_formatted_timestamp_or_none`

Reply:
```python
'happened': get_formatted_timestamp_or_none(manuscript_version['under_review_timestamp'])
```

Reply:
I will add this in one of the next PR as it will needs be changed in current DocMaps also.